core/types/block_test.go

```diff
@@ -76,6 +76,266 @@ func TestBlockEncoding(t *testing.T) {
+	// // f90202f901fda00000000000000000000000000000000000000000000000000000000000000000a00000000000000000000000000000000000000000000000000000000000000000948888f1f195afa192cfee860698584c030f4c9db1a0ef1552a40b7165c3cd773806b9e0c165b75356e0314bf0706f279c729f51e017a00000000000000000000000000000000000000000000000000000000000000000a00000000000000000000000000000000000000000000000000000000000000000b9010000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000010402030380a0bd4472abb6659ebe3ee06ee4d7b72a00a9f4d001caca51342001075469aff49888000000000000000005cec20201c20180c3028001c3018080c0c0
+	// // getting the same value in bor
+
+	var block Block
+
+	if err := rlp.DecodeBytes(blockEnc, &block); err != nil {
+		t.Fatal("decode error: ", err)
+	}
+
+	check := func(f string, got, want interface{}) {
+		if !reflect.DeepEqual(got, want) {
+			t.Errorf("%s mismatch: got %v, want %v", f, got, want)
+		}
+	}
+
+	check("Difficulty", block.Difficulty(), big.NewInt(1))
+	check("GasLimit", block.GasLimit(), uint64(2))
+	check("GasUsed", block.GasUsed(), uint64(3))
+	check("Coinbase", block.Coinbase(), libcommon.HexToAddress("8888f1f195afa192cfee860698584c030f4c9db1"))
+	check("MixDigest", block.MixDigest(), libcommon.HexToHash("bd4472abb6659ebe3ee06ee4d7b72a00a9f4d001caca51342001075469aff498"))
+	check("Root", block.Root(), libcommon.HexToHash("ef1552a40b7165c3cd773806b9e0c165b75356e0314bf0706f279c729f51e017"))
+	check("Time", block.Time(), uint64(3))
+	check("Number", block.Number(), big.NewInt(4))
+	check("BaseFee", block.BaseFee(), big.NewInt(5))
+	check("TxDependency", block.TxDependency(), [][]uint64{{2, 1}, {1, 0}, {2, 0, 1}, {1, 0, 0}})
+}
+
+func TestOnlyTxDepHeader(t *testing.T) {
+	header := &Header{
+		TxDependency: [][]uint64{{2, 0, 1}},
+	}
+
+	headerEnc, err := rlp.EncodeToBytes(&header)
+	if err != nil {
+		t.Fatal("encode error: ", err)
+	}
+
+	var headerNew Header
+	if err := rlp.DecodeBytes(headerEnc, &headerNew); err != nil {
+		t.Fatal("decode error: ", err)
+	}
+
+	check := func(f string, got, want interface{}) {
+		if !reflect.DeepEqual(got, want) {
+			t.Errorf("%s mismatch: got %v, want %v", f, got, want)
+		}
+	}
+	check("Difficulty", headerNew.TxDependency, header.TxDependency)
+}
+
+func TestOnlyTxDepBody(t *testing.T) {
+
+	var block Block
+
+	header := &Header{
+		TxDependency: [][]uint64{{2, 0, 1}, {1, 0, 0}},
+	}
+
+	block.header = header
+
+	_, err := rlp.EncodeToBytes(&block)
+	if err != nil {
+		t.Fatal("encode error: ", err)
+	}
+}
+
```

Comment:
Could you add a test case that verifies tx dependency is nil after decoding a regular tx?

Reply:
Will look into this, thanks!

Reply:
Is [this](https://github.com/maticnetwork/erigon/pull/18/commits/1216b1304aa621b4af6eb2a1bdb4a7c408a11524) okay?

Reply:
Looks good, thank you for adding it!